packages/messaging/src/index.ts

```diff
@@ -494,6 +494,7 @@ namespace MessageLoop {
    */
   const schedule = (() => {
     let ok = typeof requestAnimationFrame === 'function';
+    // @ts-ignore setImmediate is valid for NodeJS
```

Comment:
Could this be resolved by adding a dev dependency on `@types/node`?

Reply:
I don't think so as its not a NodeJS API "thing" (its a global "window"/"this" thing).

Reply:
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/028e63335139a3a0639b538a6aab9fe3fb7cfa38/types/node/timers.d.ts#L10-L15 defines it at least.

Reply:
I believe that is a different one to the "window.setImmediate" that lumino is using (the v8 JS runtime one)

Reply:
(IOW timers.setImmediate vs GLOBAL.setImmediate)

Reply:
https://nodejs.org/api/timers.html#timers_timers

> The timer module exposes a global API for scheduling functions to be called at some future period of time. Because the timer functions are globals, there is no need to call require('timers') to use the API.

Reply:
And that shows my lack of NodeJS experience!  Let me test - default to @types/node @ v12 (LTS)?

Reply:
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/5f59f3dc7a0965fb1767cf13c670c633bbbbf0a8/types/node/globals.d.ts#L177-L181

Reply:
As the typing of `schedule`/`unschedule` is not exported, it should suffice to ensure that it compiles, so v12 seems reasonable.

Reply:
no luck (after testing) and visually inspecting the NodeJS typings its declared in the "timers" module and not globally.

There is a @types/setimmediate which should work - but do you want to go down that road?

At some point you might want a "platform" test:
```
declare const process: any;

//  @ts-ignore
export const root: any = new Function("return this;")(); //  Prevent bundlers from messing with "this"

export const isBrowser: boolean = typeof window !== "undefined" && root === window;
export const isNode: boolean = typeof process !== "undefined" && process.versions != null && process.versions.node != null;
export const isTravis: boolean = isNode && process.env != null && process.env.TRAVIS != null;
```
See:  https://github.com/hpcc-systems/Visualization/blob/master/packages/util/src/platform.ts 
  
And will be able to add TypeScript guards for things like setImmediate.

Its still pretty tricky getting a single TypeScript package to "just work" for Node and Browser especially when you get into fetch polyfills and the like...

Reply:
@GordonSmith Did you see my link to the `global.d.ts` file? Also, did you add the types dependency to `tsconfig.json`'s `types` field?

Reply:
I see that now - pushed new version with suggested change.